pontoon/sync/tasks.py

```diff
@@ -162,11 +160,26 @@ def sync_project_repo(self, project_pk, repo_pk, project_sync_log_pk, now, obsol
     for locale in locales:
         try:
             with transaction.atomic():
+                # Skip locales that have nothing to sync
+                if vcs_project.synced_locales and locale not in vcs_project.synced_locales:
+                    continue
+
```

Comment:
In addition to what the PR description says, we also skip locales that have nothing to sync